charm.go

```diff
@@ -38,7 +40,15 @@ func ReadCharm(path string) (charm Charm, err error) {
 	if err != nil {
 		return nil, err
 	}
-	return charm, nil
+
+	// Find out the charm format, to Check the metadata.  It should
+	// be one format or the other.
+	format := FormatV2
```

Comment:
What is the feasibility of `Charm.Meta()` returning an interface instead of a reference?

If we did it this way we could:
- Have `MetaV1` and `MetaV2` as implementations, all embedding a `MetaBase`, but having different `Check`.
- Determine the format and therefore the implementation in `ReadCharm`
- Remove `checkV1` and `checkV2`.